app/models/location.rb

```diff
@@ -1,16 +1,57 @@
 class Location < ApplicationRecord
-  # Search request to Location IQ API
-  def self.geosearch(query)
-    raise ArgumentError, "No query for geosearch" if query.blank?
-    raise "No API key for geosearch" unless ENV['LOCATION_IQ_API_KEY']
+  include LocationHelper
 
-    base_url = "https://us1.locationiq.com/v1/search.php?key=#{ENV['LOCATION_IQ_API_KEY']}&format=json&addressdetails=1&normalizecity=1"
-    query_url = "#{base_url}&q=#{query}"
+  # Base request to LocationIQ API
+  def self.location_api_request(endpoint_query)
+    raise "No location API key" unless ENV["LOCATION_IQ_API_KEY"]
+
+    query_url = "https://us1.locationiq.com/v1/#{endpoint_query}&key=#{ENV['LOCATION_IQ_API_KEY']}&format=json"
```

Comment:
Can you move this to a constant or at least the URL?

Reply:
Not sure if it is worth it...

Reply:
sure will change